fhir-bucket/src/main/java/com/ibm/fhir/bucket/client/FHIRBucketClient.java

```diff
@@ -489,15 +591,21 @@ public Object getUserToken(HttpContext context) {
      * @param rdr
      * @return
      */
-    private void processEntity(FhirServerResponse sr, HttpEntity entity) {
+    private void processEntity(FhirServerResponse sr, HttpEntity entity, boolean parseResource) {
 
         try {
-            try (InputStream instream = entity.getContent()) {
-                Resource resource = FHIRParser.parser(Format.JSON).parse(new InputStreamReader(instream, StandardCharsets.UTF_8));
-                sr.setResource(resource);
-            } catch (FHIRParserException e) {
-                throw new IllegalStateException(e);
-            } finally {
+            if (parseResource) {
+                try (InputStream instream = entity.getContent()) {
+                    Resource resource = FHIRParser.parser(Format.JSON).parse(new InputStreamReader(instream, StandardCharsets.UTF_8));
```

Comment:
i don't think you need to wrap the instream...our parser will do that for you
```suggestion
                    Resource resource = FHIRParser.parser(Format.JSON).parse(instream);
```

Reply:
oops, i guess this was there before as well, probalby fine either way

Reply:
done